src/tmp/SampleEnginePlugin.ts

```diff
@@ -0,0 +1,176 @@
+import * as EngineApi from '@salesforce/code-analyzer-engine-api';
+
+export class SampleEnginePlugin extends EngineApi.EnginePluginV1 {
```

Comment:
Why is this in a directory called tmp under the src directory? I was expecting plugins to be injected into the actions... and then this sample engine plugin which contains stub engines... could live inside of the test directory.

Basically, the real actions will receive a list of runtime plugins... but for testing... it the stub engine(s) should be injected.

Reply:
This is so the `EngineLoader` has a real engine that it can load. Once a real engine is implemented, I'll be deleting this.
I have it in the `src` folder so that it can be imported within `src`.

Reply:
For now the runtime engine loader can just return [] in my opinion. And all this code can just live under the test folder somewhere.

Reply:
Problem: If the runtime engine loader doesn't return any engines, then it's impossible to do any QA. We need to return _something_ right now, if only to prove that it works.

Reply:
Yes QA would be difficult I agree. But that we can work around (possible with adding in config stuff right now). But ideally, the stubs should be in the test directory and passed in. If you don't do it now... you'll need to do it later anyway. Anyway, up to you.

Reply:
Clarification: The `SampleEnginePlugin` used by `src/loaders/EngineLoader` is separate from the stub engine plugins used for testing. This is purely so that the action has an engine that it can use while we wait for real engines to be implemented.
Re: engine injection: my understanding was that we were just going to hard-code it to use all the engines that we've written, and dynamic engine injection is something for the long-term. Hence the existence of the `EngineLoader` class in the first place. If we want to do dynamic engine injection based on the config, then I can do that. But I figure it might make sense to wait until we have at least one actual engine?